client/galaxy/scripts/mvc/workflow/workflow-terminals.js

```diff
@@ -455,7 +455,11 @@ var InputCollectionTerminal = BaseInputTerminal.extend({
             let node = this.node;
             _.each(node.output_terminals, function(output_terminal) {
                 if (output_terminal.attributes.collection_type_source && !connector.dragging) {
-                    output_terminal.attributes.collection_type = other.attributes.collection_type;
+                    if (other.isMappedOver()) {
+                        output_terminal.attributes.collection_type = other.terminalMapping.mapOver.collectionType;
```

Comment:
Does this work if you're mapping over a collection output type? ... and if there is subcollection mapping on the inputs? There is a lot of cases to handle here but this change is at very least a step in the right direction so I'm definitely 👍. There are some client unit test cases trying to capture the map/reduce logic for these things - I should probably update those with collection_type_source outputs to simulate some of these more complicated cases - but the tests are a bit of a mess.

Reply:
I haven't tried that yet, it might not. Good point about the tests, I'm essentially flying blind here.

Reply:
You're right, in that case I think it should be
```
other.terminalMapping.mapOver.append(other.collectionType)
```